DNN Platform/Library/Services/FileSystem/FolderManager.cs

```diff
@@ -2009,23 +2009,26 @@ private IFolderInfo GetParentFolder(int portalId, string folderPath)
 
         private IEnumerable<IFileInfo> SearchFiles(IFolderInfo folder, Regex regex, bool recursive)
         {
-            var fileCollection =
-                CBO.Instance.FillCollection<FileInfo>(DataProvider.Instance().GetFiles(folder.FolderID, false, false));
+            // Get all files in one DB call
+            var fileCollection = CBO.Instance.FillCollection<FileInfo>(
+                DataProvider.Instance().GetFiles(folder.FolderID, false, recursive));
 
-            var files = (from f in fileCollection where regex.IsMatch(f.FileName) select f).Cast<IFileInfo>().ToList();
-
-            if (recursive)
+            if (!recursive)
             {
-                foreach (var subFolder in this.GetFolders(folder))
-                {
-                    if (FolderPermissionController.Instance.CanViewFolder(subFolder))
-                    {
-                        files.AddRange(this.SearchFiles(subFolder, regex, true));
-                    }
-                }
+                return fileCollection.Where(f => regex.IsMatch(f.FileName)).Cast<IFileInfo>();

```

Comment:
This method used to return a `List` and now it returns an `IEnumerable`. In the usages within DNN itself, this appears to be a performance boost, since the two callers do sorting and then materialize the results into a list (thus, this change avoids materializing the list an extra time).

It's possible that 3rd party usages of this method will be less performant if there were reusing the returned collection, but I think the benefit it higher than the drawback.